src/pages/iou/SplitBillDetailsPage.tsx

```diff
@@ -0,0 +1,178 @@
+    report: OnyxEntry<Report>;
+
+    /** Array of report actions for this report */
+    reportActions: OnyxEntry<ReportActions>;
+
+    /** Session info for the currently logged in user. */
+    session: OnyxEntry<Session>;
+};
+
+type SplitBillDetailsPageOnyxProps = SplitBillDetailsPageTransactionOnyxProps & SplitBillDetailsPageOnyxPropsWithoutTransaction;
+
+type SplitBillDetailsPageProps = WithReportAndReportActionOrNotFoundProps &
+    SplitBillDetailsPageOnyxProps &
+    StackScreenProps<SplitDetailsNavigatorParamList, typeof SCREENS.SPLIT_DETAILS.ROOT>;
+
+function SplitBillDetailsPage({personalDetails, report, route, reportActions, transaction, draftTransaction, session}: SplitBillDetailsPageProps) {
+    const styles = useThemeStyles();
+    const reportID = report?.reportID ?? '';
+    const {translate} = useLocalize();
+    const reportAction = useMemo(() => reportActions?.[route.params.reportActionID] ?? ({} as ReportAction), [reportActions, route.params.reportActionID]);
+    const participantAccountIDs = reportAction?.actionName === CONST.REPORT.ACTIONS.TYPE.IOU ? reportAction?.originalMessage.participantAccountIDs ?? [] : [];
+
+    // In case this is workspace split bill, we manually add the workspace as the second participant of the split bill
+    // because we don't save any accountID in the report action's originalMessage other than the payee's accountID
+    let participants: Array<Participant | ReportUtils.OptionData>;
+    if (ReportUtils.isPolicyExpenseChat(report)) {
+        participants = [
+            OptionsListUtils.getParticipantsOption({accountID: participantAccountIDs[0], selected: true, reportID: ''}, personalDetails),
+            OptionsListUtils.getPolicyExpenseReportOption({...report, selected: true, reportID}),
+        ];
+    } else {
+        participants = participantAccountIDs.map((accountID) => OptionsListUtils.getParticipantsOption({accountID, selected: true, reportID: ''}, personalDetails));
+    }
+    const payeePersonalDetails = personalDetails?.[reportAction?.actorAccountID ?? -1];
+    const participantsExcludingPayee = participants.filter((participant) => participant.accountID !== reportAction?.actorAccountID);
+
+    const isScanning = TransactionUtils.hasReceipt(transaction) && TransactionUtils.isReceiptBeingScanned(transaction);
+    const hasSmartScanFailed = TransactionUtils.hasReceipt(transaction) && transaction?.receipt?.state === CONST.IOU.RECEIPT_STATE.SCANFAILED;
+    const isEditingSplitBill = session?.accountID === reportAction?.actorAccountID && TransactionUtils.areRequiredFieldsEmpty(transaction);
+
+    const {
+        amount: splitAmount,
+        currency: splitCurrency,
+        comment: splitComment,
+        merchant: splitMerchant,
+        created: splitCreated,
+        category: splitCategory,
+        tag: splitTag,
+        billable: splitBillable,
+    } = ReportUtils.getTransactionDetails(isEditingSplitBill ? draftTransaction : transaction) ?? {};
```

Comment:
```suggestion
    } = ReportUtils.getTransactionDetails(isEditingSplitBill && draftTransaction ? draftTransaction : transaction) ?? {};
```

Earlier logic here
https://github.com/Expensify/App/blob/00e0121fa17a7c19335a8fa10eed51c25e933d6a/src/pages/iou/SplitBillDetailsPage.js#L106
used `transaction` for the `getTransactionDetails` when `isEditingSplitBill` is `true` but `draftTransaction` is `false` (`undefined`). Now, with the present logic we will use `draftTransaction` when `isEditingSplitBill` is `true` even when `draftTransaction` might be `undefined`.

Reply:
Great catch @c3024 ! Thank you, I'll apply it 😄